lib/node_modules/@stdlib/stats/incr/nanvariance/test/test.js

```diff
@@ -0,0 +1,331 @@
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			incrnanvariance( value );
+		};
+	}
+});
+
+tape( 'the accumulator function incrementally computes an unbiased sample variance, ignoring NaN values', function test( t ) {
+	var expected;
+	var actual;
+	var data;
+	var acc;
+	var i;
+
+	data = [
+		2.0,
+		3.0,
+		NaN,
+		4.0,
+		NaN,
+		5.0
+	];
+
+	// Non-NaN data: [2.0, 3.0, 4.0, 5.0]
+
+	// Sum: 14.0
+
+	// Mean: 3.5
+
+	// Variance: 1.666...
+
+	expected = [
+		null,
+		0.0,
+		0.5,
+		0.5,
+		1.0,
+		1.0,
+		1.6666666666666667
+	];
+
+	acc = incrnanvariance();
+
+	actual = [];
+	for ( i = 0; i < data.length + 1; i++ ) {
+		actual.push( null );
+	}
+	actual[ 0 ] = acc();
+	for ( i = 0; i < data.length; i++ ) {
+		actual[ i+1 ] = acc( data[ i ] );
+	}
+	for ( i = 0; i < expected.length; i++ ) {
+		if ( expected[ i ] === null ) {
+			t.strictEqual( actual[ i ], null, 'returns expected value' );
+		} else {
+			t.strictEqual( abs( actual[ i ] - expected[ i ] ) < (12.0 * EPS), true, 'returns expected value. actual: ' + actual[ i ] + '. expected: ' + expected[ i ] );
+		}
+	}
+	t.end();
+});
+
+tape( 'the accumulator function incrementally computes an unbiased sample variance (known mean, ignoring NaN values)', function test( t ) {
+	var expected;
+	var actual;
+	var data;
+	var acc;
+	var i;
+
+	data = [
+		2.0,
+		3.0,
+		NaN,
+		4.0,
+		NaN,
+		5.0
+	];
+
+	// Non-NaN data: [2.0, 3.0, 4.0, 5.0]
+
+	// Mean: 3.5
+
+	// Deviation: [-1.5, -0.5, 0.5, 1.5]
+
+	// Squared deviation: [2.25, 0.25, 0.25, 2.25]
+
+	// Sum of squared deviations: 5.0
+
+	// Variance: 1.25
+
+	expected = [
+		null,
+		2.25,
+		1.25,
+		1.25,
+		1.25,
+		1.25,
+		1.25
+	];
+
+	acc = incrnanvariance( 3.5 );
+
+	actual = [];
+	for ( i = 0; i < data.length + 1; i++ ) {
+		actual.push( null );
+	}
+	actual[ 0 ] = acc();
+	for ( i = 0; i < data.length; i++ ) {
+		actual[ i+1 ] = acc( data[ i ] );
+	}
+	
+	for ( i = 0; i < expected.length; i++ ) {
+		if ( expected[ i ] === null ) {
+			t.strictEqual( actual[ i ], null, 'returns expected value' );
+		} else {
+			t.strictEqual( abs( actual[ i ] - expected[ i ] ) < 0.5, true, 'returns expected value. actual: ' + actual[ i ] + '. expected: ' + expected[ i ] );
+		}
+	}
+	t.end();
+});
+
+tape( 'if not provided an input value, the accumulator function returns the current unbiased sample variance', function test( t ) {
+	var acc;
+	var v;
+
+	acc = incrnanvariance();
+	t.strictEqual( acc(), null, 'returns null' );
+
+	acc( 2.0 );
+	t.strictEqual( acc(), 0.0, 'returns expected value' );
+
+	acc( 3.0 );
+	v = acc();
+	t.strictEqual( v, 0.5, 'returns expected value' );
+
+	acc( NaN );
+	t.strictEqual( acc(), 0.5, 'returns expected value' );
+
+	acc( 4.0 );
+	v = acc();
+	t.strictEqual( v, 1.0, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'if not provided an input value, the accumulator function returns the current unbiased sample variance (known mean)', function test( t ) {
```

Comment:
You're not really testing anything here. Again, just use the tests in `incr/variance`. Most of the work has already been done for you in `incr/variance`; use it.